src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -0,0 +1,275 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Security.Cryptography;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+#if NET472
```

Comment:
We need to add an additional target that can be used from .Net core.
We could add 5.0 and close this issue.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1426

Reply:
Question: Should I make the additional target part of this PR?

Comment:
There are two referenced classes that have potential conflicts with versioning in this file:
- [ECParameters](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.ecparameters?view=net-6.0) is available in all Core versions, and .NET Framework 4.7 onwards
- [ECDiffieHellman](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.ecdiffiehellman?view=net-6.0) is available in Core 2.1 onwards, and .NET Framework 3.5 onwards

Would it look something like?
`#if NET472_OR_GREATER || NETSTANDARD2_1_OR_GREATER`

Reply:
Per PR Meeting, we'll add .NET Core 3.1

Reply:
Added in [f6f8794](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1834/commits/f6f8794f6309d9962e52064dd47858458ee10130)